packages/firestore/src/core/target_id_generator.ts

```diff
@@ -31,6 +31,8 @@ enum GeneratorIds {
  * will ever produce the same ID. This is useful, because sometimes the backend
  * may group IDs from separate parts of the client into the same ID space.
```

Comment:
I think this comment could better reflect the state of the world with your simplifications...  Something like:

Generates target IDs for sending targets to the watch stream. The client constructs two generators, one for the query cache (via forQueryCache()), and one for limbo documents (via forSyncEngine()).  The two generators produce non-overlapping IDs (by using even and odd IDs respectively). 

Having two separate generators is useful because query cache target IDs persist across client restarts and must not be reused, while sync engine target IDs are transient and can be reused after a restart.